src/cellars/cellar_v2_2.rs

```diff
@@ -0,0 +1,368 @@
+
+            Ok(CellarV2_2Calls::CallOnAdaptor(call).encode())
+        }
+        Function::RemovePosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionCall {
+                index: params.index,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_2Calls::RemovePosition(call).encode())
+        }
+        Function::RemoveAdaptorFromCatalogue(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemoveAdaptorFromCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemoveAdaptorFromCatalogueCall {
+                adaptor: sp_call_parse_address(params.adaptor)?,
+            };
+
+            Ok(CellarV2_2Calls::RemoveAdaptorFromCatalogue(call).encode())
+        }
+        Function::RemovePositionFromCatalogue(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionFromCatalogueCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionFromCatalogueCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2_2Calls::RemovePositionFromCatalogue(call).encode())
+        }
+        Function::SetHoldingPosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetHoldingPositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetHoldingPositionCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2_2Calls::SetHoldingPosition(call).encode())
+        }
+        Function::SetStrategistPayoutAddress(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetStrategistPayoutAddressCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetStrategistPayoutAddressCall {
+                payout: sp_call_parse_address(params.payout)?,
+            };
+
+            Ok(CellarV2_2Calls::SetStrategistPayoutAddress(call).encode())
+        }
+        Function::SwapPositions(params) => {
+            log_cellar_call(CELLAR_NAME, &SwapPositionsCall::function_name(), &cellar_id);
+            let call = SwapPositionsCall {
+                index_1: params.index_1,
+                index_2: params.index_2,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2_2Calls::SwapPositions(call).encode())
+        }
+    }
+}
+
+pub fn get_encoded_governance_call(
+    function: GovernanceFunction,
+    cellar_id: &str,
+    proposal_id: u64,
+) -> Result<Vec<u8>, Error> {
+    match function {
+        GovernanceFunction::AddAdaptorToCatalogue(params) => {
+            log_governance_cellar_call(
```

Comment:
Our governance functions here and in Cellar V2 should respect blocked adaptors and blocked positions.

Reply:
Should setupAdaptor check blocked adaptors?

Reply:
Yes, we should disallow blocked adaptors from both `setupAdaptor`/`addAdaptorToCatalogue` and `callOnAdaptor`.